### ISSUE TITLE
the overfitting problem

### ISSUE THREAD

**Issue by linhanxiao** [facebookresearch/fairseq-py#90]
i have set the weight_decay=1e-07，but the valid_loss is inreasing while the train loss is  decreasing
@edunov

**Comment by edunov** [comment]
What is the dataset? What are the other options? What exactly do you see in the log? Weight decay is not a magic bullet it will not stop overfitting if other parameters are off

**Comment by linhanxiao** [comment]
CUDA_VISIBLE_DEVICES=0 python train.py data-bin/flickr8k \
  --lr 0.33 --clip-norm 0.1 --dropout 0.2 --max-tokens 4000 \
  --arch fconv_caption --save-dir /mnt/fairseq/checkpoints/fconv_caption/flickr8k  --max-epoch 1000   --weight-decay 0.000001

**Comment by linhanxiao** [comment]
i want to solve the image caption problem using the fairseq-py code，I have modified the encoder in faiseq-py code.my dataset is flickr8k which has 8000 picture and MSCOCO which has 120000 picture @edunov

**Comment by linhanxiao** [comment]
i want to solve the image caption problem using the fairseq-py code，.my dataset is flickr8k which has 8000 picture and MSCOCO which has 120000 picture @edunov

**Comment by edunov** [comment]
Ok, can you paste the log as well? 
How did you split data into train and valid? The biggest question here is is valid drawn from the same distribution as train? Did you shuffle data before splitting? Is all the preprocessing the same for valid and train?  
If the valid and the train are set properly, then there might be something wrong with the model, you're welcome to submit pull request with your changes, maybe we can spot something. 

Another thing to try would be to reduce the model size, e.g. reduce the number of hidden layers. If you make a very small model, will it still overfit?

**Comment by linhanxiao** [comment]
We use the Flickr8K  and MSCOCO  datasets in our experiments. These datasets
contain 8,000, 123,000 images respectively and each is annotated with 5 sentences,For Flickr8K and Flickr30K, we use 1,000 images for validation, 1,000 for testing and the rest for training . For MSCOCO we use 5,000 images for both validation and testing. all the preprocessing is the same for valid and train.I have made a very small model ,it will underfit

**Comment by linhanxiao** [comment]
@edunov

**Comment by linhanxiao** [comment]
We use the Flickr8K and MSCOCO datasets in our experiments. These datasets
contain 8,000, 123,000 images respectively and each is annotated with 5 sentences,For Flickr8K and Flickr30K, we use 1,000 images for validation, 1,000 for testing and the rest for training . For MSCOCO we use 5,000 images for both validation and testing. all the preprocessing is the same for valid and train.I have made a very small model ,it will underfit @edunov

**Comment by linhanxiao** [comment]
i have extracted image feature with VGG neural network, the x and y in the code line 'return x, y' in encoder forward function are all image feature. @huihuifan

**Comment by linhanxiao** [comment]
I have extract image feature with VGG  CNN,the encoder output x and y are all the image feature @huihuifan
